core/app/models/spree/product.rb

```diff
@@ -161,6 +161,11 @@ def discontinued?
       !!discontinue_on && discontinue_on <= Time.current
     end
 
+    # determine if any variant (including master) can be supplied
+    def can_supply?
+      variants_including_master.any?(&:can_supply?)
```

Comment:
@Mafi88 When a variant is created (which is not master) then the master's count on hand is reduced to 0. https://github.com/spree/spree/blob/master/core/app/models/spree/variant.rb#L48.
But having a check here seems fine. Like if variants are present then check supply of variants else check supply of master.